The first commit with generics OPEN

Flottika opened the pull request:
The first commit with generics (class & interface)

sbuvaka-fhl committed:
Add dev branch for Y.Practicum

Flottika committed:
the first commit

Flottika commented:
The first commit with generics

faritowich commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java:
```diff
@@ -0,0 +1,29 @@
+
+    public boolean theSameAs(T newItem) {
+        return this.hashCode() == newItem.hashCode();
+    }
```
Этот метод лишний и не нужен

Flottika commented:
Добрый вечер!
Спасибо за ревью. Да, насчет этого не была уверена, что стоит здесь писать.
Правильно я понимаю, тк..класс BaseListDiffCallback
не являеся расширением интерфейса ListDiffInterface, то  можно не
реализоваывать там метод  theSameAs (T newItem)?

У меня есть несколько вопросов по ревью:

   1. ❌ Создан интерфейс ListDiffInterface с двумя методами theSameAs() и
   equals()
   2.
   3.

   public interface ListDiffInterface <T> {
       boolean theSameAs(T t);
       boolean equals(Object o);
   }

   Здесь же создан интерфейс и объявлены методы?
   4. ❌ Создан дженерик-класс BaseListDiffCallback с реализацией
   5.
   6. Здесь это сообщение из-за метода theSameAs?
   7.
   8. Нужно чтобы было так:
   9.

   @Override

   10.

   public boolean ***@***.*** T oldItem, @NonNull T newItem) {

   11.

       return ((ListDiffInterface<T>) this).theSameAs(newItem);
   }


- Неправильно написан интерфейс ListDiffCallback - в нем сейчас не
объявлены никакие методы
В задании не было указаний делать что-то с интерфейсом ListDiffCallback,
правильно ли я понимаю, что ошибка в том, что он остался в коде и его нужно
просто удалить ?

вт, 11 июн. 2024 г. в 14:18, Azat Nabiullin ***@***.***>:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In
> app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java
> <https://github.com/Flottika/android-contacts/pull/1#discussion_r1634589093>
> :
>
> > +
> +import androidx.annotation.NonNull;
> +import androidx.annotation.Nullable;
> +
> +import androidx.recyclerview.widget.DiffUtil;
> +
> +public class BaseListDiffCallback <T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T> {
> +
> +    @Override
> +    public boolean ***@***.*** T oldItem, @NonNull T newItem) {
> +        return theSameAs(newItem);
> +    }
> +
> +    public boolean theSameAs(T newItem) {
> +        return this.hashCode() == newItem.hashCode();
> +    }
>
> Этот метод лишний и не нужен
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Flottika/android-contacts/pull/1#pullrequestreview-2109992715>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BBWGV74VTYMHUJEJ7JXLQ2TZG3FITAVCNFSM6AAAAABJBFQ3B2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDCMBZHE4TENZRGU>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>

Flottika committed:
the second commit:
deleted method theSameAs at BaseListDiffCallback and changed method in BaseListDiffCallback:
 @Override
    public boolean areItemsTheSame(@NonNull T oldItem, @NonNull T newItem) {
        return oldItem.theSameAs(newItem);
    }

Flottika committed:
the second commit:
deleted method theSameAs at BaseListDiffCallback and changed method in BaseListDiffCallback:
 @Override
    public boolean areItemsTheSame(@NonNull T oldItem, @NonNull T newItem) {
        return oldItem.theSameAs(newItem);
    }

faritowich commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/base/ListDiffInterface.java:
```diff
@@ -0,0 +1,9 @@
+
+public interface ListDiffInterface <T> {
+
+    boolean theSameAs(T t);
```
Лучше использовать такой нейминг, чтобы по названию было понятно, что параметр означает. Например, anotherItem вместо t
